bin/porta

```diff
@@ -330,6 +334,32 @@ module Porta
 
     protected
 
+    def docker_internal_host
+      options[:docker_internal_host] ||= containe_local_host_ip
+    end
+
+    def containe_local_host_ip(image: nil)
+      image ||= options[:porxy_image]
```

Comment:
I'd avoid mutating the state in general.

Reply:
This is only a local variable. Or you mean next line?

 Next line is to cache default gw ip. Otherwise we may start a new container up to 4 times just to get that same value. If it is cached, that is performed only once. If you confirm, I'll remove this caching.

Reply:
This particular case is harmless, but try this in the Ruby terminal:

```ruby
irb(main):001:0> def m(s: nil)
irb(main):002:1>   s ||= 'default-value'
irb(main):003:1>   s << '-suffix'
irb(main):004:1>   s
irb(main):005:1> end
=> :m

irb(main):006:0> m
=> "default-value-suffix"

irb(main):007:0> m(s: 'other-value')
=> "other-value-suffix"

irb(main):008:0> v = 'closure'
=> "closure"

irb(main):009:0> m(s: v)
=> "closure-suffix"

irb(main):010:0> v
=> "closure-suffix" # String#<< mutates the state of the string
```

I prefer always avoiding operations that mutate the state of the input parameters. They can lead to unpredictable results in the outside of the method. Instead, when I really want to mutate the state, I usually define a class and implement the mutating method as an instance method; and for all other cases, I copy the value of the parameter into a local variable with a different name.